pkg/tsdb/azuremonitor/azure-resource-graph-datasource_test.go

```diff
@@ -110,3 +113,34 @@ func TestAzureResourceGraphCreateRequest(t *testing.T) {
 		})
 	}
 }
+
+func TestAddConfigData(t *testing.T) {
+	field := data.Field{}
+	dataLink := data.DataLink{Title: "View in Azure Portal", TargetBlank: true, URL: "http://ds"}
+	frame := data.Frame{
+		Fields: []*data.Field{&field},
+	}
+	frameWithLink := addConfigData(frame, "http://ds")
+	assert.NotEmpty(t, frameWithLink.Fields)
+	assert.NotNil(t, frameWithLink.Fields[0].Config)
+	assert.NotEmpty(t, frameWithLink.Fields[0].Config.Links)
+	assert.Equal(t, frameWithLink.Fields[0].Config.Links[0], dataLink)
```

Comment:
it's a bit more readable (and easier to discover errors) if you compare the structs with `cmp.Diff`:

```
expectedFrameWithLink := &{...}
if !cmp.Equal(frameWithLink, expectedFrameWithLink) {
  t.Errorf("unexpepcted frame: %v", cmp.Diff(frameWithLink, expectedFrameWithLink))
}
```

That way, with one line, you get all the unexpected differences in a diff view.

Reply:
Done.